### PR TITLE
Add chat server

### LINKED ISSUE #125: So that event participants can find our discord server, as an event host, I want to be able to add a link to a discord or other server
- [x] Event hosts can update
- [x] Not required
- [x] chat server name field
- [x] chat server URL
- [x] chat server URL is required if chat server isn't blank
- [x] Should display under location on event page if not blank. 
- [x] Should not display if blank

### PR DESCRIPTION
**Pull request by Silbeg**
This adds upon PR #139 by @blomquistr by refactoring  `Event#valid_chat_server` under green.

- Also adding tests with empty spaces, which also still pass.
- Plus some minor reformatting
- Added missing `event#chat_server?`
- Updated seeds to use dynamic date on seeding
- Refactor under green for clarity

----
This was done for clarity, to remove the nested conditionals and to use some more standard Rails checks (`string.blank?` and `string.present?` that someone not familiar with Rails might not know about.  In fact, I had forgotten that `string.present?`  even exists!

See: https://gist.github.com/pythonicrubyist/8114720 for details on what these all do.

Also added specs to test for whitesprace 

I find this code to be a lot cleaner to read, personally. 

Fixes #125

### CODE REVIEW

File: spec/model/event_spec.rb [hunk 1]
```diff
@@ -35,4 +35,85 @@
...
+
+    it 'If event has a chat_server_url but no chat_server error should exist' do
+      event = Event.new
+      event.chat_server = nil
+      event.chat_server_url = 'https://www.google.com'
+
+      event.save
+      expect(event.valid_chat_server).to be false
+      expect(event.errors[:chat_server]).to_not be_empty
+    end
```
Comment (Silbeg) [inline spec/model/event_spec.rb:67]: @blomquistr I actually prefer this rule to be implemented. While we are suppressing the display if `chat_server` is nil (reminds me, should change that to blank), this prevents someone from getting confused why their chat_server_url isn't displaying when they didn't give a chat server.
